mmv1/products/certificatemanager/IssuanceConfigs.yaml

```diff
@@ -0,0 +1,127 @@
+  - !ruby/object:Api::Type::Integer
+    name: 'rotationWindowPercentage'
+    description: |
+      It specifies the percentage of elapsed time of the certificate lifetime to wait before renewing the certificate. 
+      Must be a number between 1-99, inclusive. Defaults to 66. 
+      You must set the rotation window percentage in relation to the certificate lifetime so that certificate renewal occurs at least 7 days after 
+      the certificate has been issued and at least 7 days before it expires.
+  - !ruby/object:Api::Type::Enum
+    name: keyAlgorithm
+    description: |
+      Key algorithm to use when generating the private key. Defaults to
+      rsa-2048. KEY_ALGORITHM must be one of: ECDSA-P256, RSA_2048.
+    default_value: :RSA_2048
+    values:
+      - :RSA_2048
+      - :ECDSA_P256
+  - !ruby/object:Api::Type::String
+    name: 'lifetime'
+    description: |
+      Lifetime of issued certificates. A duration in seconds with up to nine fractional digits, ending with 's'. 
+      Example: "1814400s". Defaults to 2592000s (30 days). Valid values are from 21 days (1814400s) to 30 days (2592000s)
+    default_value: 2592000s
+  - !ruby/object:Api::Type::String
+    name: 'createTime'
+    description: |
+      The creation timestamp of a CertificateIssuanceConfig. Timestamp is in RFC3339 UTC "Zulu" format,
+      accurate to nanoseconds with up to nine fractional digits.
+      Examples: "2014-10-02T15:01:23Z" and "2014-10-02T15:01:23.045123456Z".
+    output: true
+  - !ruby/object:Api::Type::Time
+    name: 'updateTime'
+    description: |
+      The last update timestamp of a CertificateIssuanceConfig. Timestamp is in RFC3339 UTC "Zulu" format,
+      accurate to nanoseconds with up to nine fractional digits.
+      Examples: "2014-10-02T15:01:23Z" and "2014-10-02T15:01:23.045123456Z".
+    output: true
+  - !ruby/object:Api::Type::KeyValuePairs
+    name: 'labels'
+    description: |
+      'Set of label tags associated with the CertificateIssuanceConfig resource. 
+       An object containing a list of "key": value pairs. Example: { "name": "wrench", "count": "3" }.
+  - !ruby/object:Api::Type::NestedObject
+    name: 'certificateAuthorityConfig'
+    description: |
+      The CA that issues the workload certificate. It includes the CA address, type, authentication to CA service, etc.
+    required: true
+    properties:
+      - !ruby/object:Api::Type::NestedObject
+        name: 'certificateAuthorityServiceConfig'
+        description: Defines a CertificateAuthorityServiceConfig.
+        properties: 
+          - !ruby/object:Api::Type::String
+            name: 'caPool'
+            required: true
+            description: |
+              A CA pool resource used to issue a certificate. 
+              The CA pool string has a relative resource path following the form 
+              "projects/{project}/locations/{location}/caPools/{caPool}".
+            diff_suppress_func: 'tpgresource.CompareResourceNames' 
```

Comment:
I found out that after terraform creates the ca-pool, it gets a response with a different caPool name. or example, if the caPool is:
`projects/{{project_id}}/locations/{{location}}/caPools/{{caPoolName}}`
I receive a response with caPool equals:
`projects/{{project_number}}/locations/{{location}}/caPools/{{caPoolName}}`

Both names are equivalent despite not being identical, so I added this suppress function to handle this situation. 

Is their a better way to do it?

Reply:
Ideally the API wouldn't do that, as it violates the AIPS: https://google.aip.dev/cloud/2510#resource-references

Otherwise, yes, we've got to use a diffsuppressfunc which may result in false negatives (where there is drift, but Terraform does not detect it)